mantidproject/mantid
# Results table for MuonAnalysis: run number and period label

The MuonAnalysis interface produces a results table in which one of the columns that can be selected is the run number.
This PR makes two changes to this column:
1. In the case of co-added runs, all runs are put in the label, not just the first one (i.e. `15189-91` rather than just `15189`)
2. The period, or combination of periods, used has been added to the label, for example `15189: 1` or `15189-91: 1+2-3+4`. The period label is left off in the case of single-period data, or if all periods are used.

When fitting more than one period from the same run, previously there was only one entry in the results table - now each fit should have its own entry.

**To test:**

(Unfortunately there are no automated tests for this interface.)
_Example data files: `MUSR00015189-91.nxs` from the doc tests folder_
1. Add the data files to Mantid's search path and start the MuonAnalysis interface
2. Set instrument to MUSR and type "15189-91" in the box to load these runs
3. Data Analysis tab: fit something to the data
4. Go back to the home tab and change the periods analysed - use period 2, or 1+2, 1-2...
5. Fit this again, repeat a few times with different data combinations
6. Results Table tab: generate table and check that the run number/period labels are as expected
7. Try again with a single-period dataset, or loading all periods from a multi-period set

Fixes #15537 

[Release notes](https://github.com/mantidproject/mantid/blob/f0df4613ce67ffdce723a9f266bc72d1dc66aa94/docs/source/release/v3.7.0/muon.rst) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
